### PR TITLE
docs(darkmode): add demo and Timeline content add color scss variable

### PR DESCRIPTION
**Pull request by shijiatongxue**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [x] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [x] Document improve

### PR description
- Add block dark mode demo @pointhalo 
- Timeline content add color scss variable(prevent inheritance from body) @pointhalo 
- Remove `data_client.json` fs.copyFileSync used by doc search @DaiQiangReal please check this

### Usage

image

### Demo

block-dark-mode3

### CODE REVIEW

File: content/start/dark-mode/index.md [hunk 1]
```diff
@@ -27,14 +27,7 @@ if (body.hasAttribute('theme-mode')) {
...
 
-class Demo extends React.Component {
-    constructor() {
-        super();
-        // 为了通知官网记录当前模式，这里引入了监听模式变化
-        // 不同的业务场景可能会采取不同方式，请留意
-    }
-
-    switchMode() {
+class Demo extends React.Component {    switchMode() {
```
Comment (pointhalo) [inline content/start/dark-mode/index.md:30]: 这行format一下

File: content/start/dark-mode/index.md [hunk 2]
```diff
@@ -45,6 +38,13 @@ class Demo extends React.Component {
             window.setMode('dark');
         }
     }
+    constructor() {
+        super();
+        // 为了通知官网记录当前模式，这里引入了监听模式变化
+        // 不同的业务场景可能会采取不同方式，请留意
+    }
+
+
```
Comment (pointhalo) [inline content/start/dark-mode/index.md:47]: 这个constuctor是不是去掉

File: content/start/dark-mode/index.md (hunk 1 above)
Comment (shijiatongxue) [reply content/start/dark-mode/index.md:30]: 误改了这里，我恢复下

File: content/start/dark-mode/index.md (hunk 2 above)
Comment (shijiatongxue) [reply content/start/dark-mode/index.md:47]: 我处理下

**Pull request merged**